src/S3/S3Client.php

```diff
@@ -239,80 +239,105 @@ class S3Client extends AwsClient implements S3ClientInterface
                 ],
-                'doc'     => 'Set to true to allow passed in ARNs to override'
-                    . ' client region. Accepts...',
-                'fn' => [__CLASS__, '_apply_use_arn_region'],
-                'default' => [UseArnRegionConfigurationProvider::class, 'defaultProvider'],
-            ],
-            'use_accelerate_endpoint' => [
-                'type' => 'config',
-                'valid' => ['bool'],
-                'doc' => 'Set to true to send requests to an S3 Accelerate'
-                    . ' endpoint by default. Can be enabled or disabled on'
-                    . ' individual operations by setting'
-                    . ' \'@use_accelerate_endpoint\' to true or false. Note:'
-                    . ' you must enable S3 Accelerate on a bucket before it can'
-                    . ' be accessed via an Accelerate endpoint.',
-                'default' => false,
-            ],
-            'use_path_style_endpoint' => [
-                'type' => 'config',
-                'valid' => ['bool'],
-                'doc' => 'Set to true to send requests to an S3 path style'
-                    . ' endpoint by default.'
-                    . ' Can be enabled or disabled on individual operations by setting'
-                    . ' \'@use_path_style_endpoint\' to true or false.',
-                'default' => false,
-            ],
-            'disable_multiregion_access_points' => [
-                'type' => 'config',
-                'valid' => ['bool'],
-                'doc' => 'Set to true to disable the usage of'
-                    . ' multi region access points. These are enabled by default.'
-                    . ' Can be enabled or disabled on individual operations by setting'
-                    . ' \'@disable_multiregion_access_points\' to true or false.',
-                'default' => false,
-            ],
-            'disable_express_session_auth' => [
-                'type' => 'config',
-                'valid' => ['bool'],
-                'doc' => 'Set to true to disable the usage of'
-                    . ' s3 express session authentication. This is enabled by default.',
-                'default' => [__CLASS__, '_default_disable_express_session_auth'],
-            ],
-            's3_express_identity_provider' => [
-                'type'    => 'config',
-                'valid'   => [
-                    'bool',
-                    'callable'
+                'response_checksum_validation' => [
+                    'type' => 'config',
+                    'valid' => ['string'],
+                    'doc' => 'Set to true to disable the usage of'
+                        . ' s3 express session authentication. This is enabled by default.',
+                    'fn' => [__CLASS__, '_apply_response_checksum_validation'],
+                    'default' => [__CLASS__, '_default_response_checksum_validation'],
+                ]
+            ]
+            + $args + [
```

Comment:
Question: Why not concatenate the `$args` at the end or at the beginning? We could miss this later since it is sort of hidden.

Reply:
One of these two options (`response_checksum_validation`) needs to be prepended to the list because its value must be resolved before [`apply_api_provider` is called](https://github.com/aws/aws-sdk-php/blob/master/src/ClientResolver.php#L122,#128), due to the fact that `ValidateResponseCheckSumMutator` is applied to the api provider.  I suppose we could just do this with `response_checksum_validation` and move `request_checksum_calcuation` into the larger list of s3 specific config. What do you think?